build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/GitBranch.cs

```diff
@@ -21,18 +21,21 @@ public sealed class GitBranch : Git
 			get { return false; }
 		}
 
-		public GitBranch ()
-		{
-			// `git rev-parse --abbrev-ref HEAD` currently writes `HEAD` to stderr; no idea why.
-			LogStandardErrorAsError = false;
-		}
-
 		public override bool Execute ()
 		{
 			Log.LogMessage (MessageImportance.Low, $"Task {nameof (GitBranch)}");
 			Log.LogMessage (MessageImportance.Low, $"  {nameof (WorkingDirectory)}: {WorkingDirectory.ItemSpec}");
 
-			base.Execute ();
+			string gitHeadFile = Path.Combine (WorkingDirectory.ItemSpec, ".git", "HEAD");
+			if (File.Exists (gitHeadFile)) {
+				string gitHeadFileContent = File.ReadAllText (gitHeadFile);
+				Match match = Regex.Match (gitHeadFileContent, "(?<=refs/heads/).+$");
```

Comment:
We could declare this `Regex` object as a `static readonly` field and use `RegexOptions.Compiled`. Generally using the static `Regex.Match` is pretty slow, but OK if only called once.

The `GetXAVersionInfo` target runs a lot... So it's worth doing it, I think.

Reply:
@jonathanpeppers The suggested perf change has been applied 👍